pkg/rabbitmq/producer.go

```diff
@@ -70,11 +70,12 @@ func (p *Producer) Produce(channel Channel, payload any) {
 		publishing)
 	failOnError(err, "Failed to publish a message")
 
-	log.Printf("[%s] Sent %+v", channel, payload)
+	p.logger.Info("message produced", "channel", channel, "payload", payload)
 }
 
 func failOnError(err error, msg string) {
 	if err != nil {
-		log.Fatalf("%s: %s", msg, err)
+		_, _ = fmt.Fprintf(os.Stderr, "%s: %s", msg, err) // nolint:errcheck
+		os.Exit(1)
```

Comment:
Since you are changing this, can you return an error instead please so we only do `os.Exit(1)` in a main.